### PR TITLE
use asanyarry in high level functions

### PR DESCRIPTION
**Pull request by hmaarrfk**
I couldn't find where the high level API tests went. I added them in core. I need to add a test for the video one too.

I'm really not sure what can't be coerced into a numpy.

Fixes:  #393

### CODE REVIEW

**Comment by hmaarrfk** [comment]
This PR is by no means complete, I’m mostly asking for your thoughts on how you think we can test for objects being coerced into numpy arrays. I’m not too sure it matters if it fails early, or later in the call to img as int.

File: imageio/core/functions.py [hunk 1]
```diff
@@ -242,17 +242,16 @@ def imwrite(uri, im, format=None, **kwargs):
...
-            pass
-        elif im.ndim == 3 and im.shape[2] in [1, 3, 4]:
-            pass
-        else:
-            raise ValueError("Image must be 2D (grayscale, RGB, or RGBA).")
+    if im.ndim == 2:
+        pass
+    elif im.ndim == 3 and im.shape[2] in [1, 3, 4]:
+        pass
     else:
```
Comment (almarklein) [inline imageio/core/functions.py]: Would be nice if this can be reduced to a single `if not (...)`

Comment (almarklein) [reply imageio/core/functions.py]: And I think I prefer to fail early here if the dtype is not numeric (or some similar measure).

**Comment by almarklein** [comment]
Nice! I think that we should accept "anything that can be coerced into a sensible (numeric and of appropriate shape) array".

**Comment by almarklein** [comment]
Ok, I added a bit to this and merged. The most important bit I added (to me) was to test that the dtype was checked to be numeric, so that when bullocks input is given, the error message is much more descriptive,

Thanks for your contribution!

**Comment by hmaarrfk** [comment]
cool things!

**Pull request closed without merging**